reference/test/preconditioner/isai_kernels.cpp

```diff
@@ -0,0 +1,626 @@
+
+    Isai()
+        : exec{gko::ReferenceExecutor::create()},
+          l_dense{gko::initialize<Dense>(
+              {{2., 0., 0.}, {1., -2., 0.}, {-1., 1., -1.}}, exec)},
+          l_dense_inv{gko::initialize<Dense>(
+              {{.5, 0., 0.}, {.25, -.5, 0.}, {-.25, -.5, -1.}}, exec)},
+          u_dense{gko::initialize<Dense>(
+              {{4., 1., -1.}, {0., -2., 4.}, {0., 0., 8.}}, exec)},
+          u_dense_inv{gko::initialize<Dense>(
+              {{.25, .125, -0.03125}, {0., -.5, .25}, {0., 0., .125}}, exec)},
+          l_csr{Csr::create(exec)},
+          l_csr_inv{Csr::create(exec)},
+          u_csr{Csr::create(exec)},
+          u_csr_inv{Csr::create(exec)},
+          l_sparse{Csr::create(exec, gko::dim<2>(4, 4),
+                               I<value_type>{-1., 2., 4., 5., -4., 8., -8.},
+                               I<index_type>{0, 0, 1, 1, 2, 2, 3},
+                               I<index_type>{0, 1, 3, 5, 7})},
+          l_s_unsorted{Csr::create(exec, gko::dim<2>(4, 4),
+                                   I<value_type>{-1., 4., 2., 5., -4., -8., 8.},
+                                   I<index_type>{0, 1, 0, 1, 2, 3, 2},
+                                   I<index_type>{0, 1, 3, 5, 7})},
+          l_sparse_inv{
+              Csr::create(exec, gko::dim<2>(4, 4),
+                          I<value_type>{-1., .5, .25, .3125, -.25, -.25, -.125},
+                          I<index_type>{0, 0, 1, 1, 2, 2, 3},
+                          I<index_type>{0, 1, 3, 5, 7})},
+          l_sparse2{Csr::create(exec, gko::dim<2>(4, 4),
+                                I<value_type>{-2, 1, 4, 1, -2, 1, -1, 1, 2},
+                                I<index_type>{0, 0, 1, 1, 2, 0, 1, 2, 3},
+                                I<index_type>{0, 1, 3, 5, 9})},
+          l_sparse2_inv{Csr::create(exec, gko::dim<2>(4, 4),
+                                    I<value_type>{-.5, .125, .25, .125, -.5,
+                                                  .28125, .0625, 0.25, 0.5},
+                                    I<index_type>{0, 0, 1, 1, 2, 0, 1, 2, 3},
+                                    I<index_type>{0, 1, 3, 5, 9})},
+          u_sparse{
+              Csr::create(exec, gko::dim<2>(4, 4),
+                          I<value_type>{-2., 1., -1., 1., 4., 1., -2., 1., 2.},
+                          I<index_type>{0, 1, 2, 3, 1, 2, 2, 3, 3},
+                          I<index_type>{0, 4, 6, 8, 9})},
+          u_s_unsorted{
+              Csr::create(exec, gko::dim<2>(4, 4),
+                          I<value_type>{-2., -1., 1., 1., 1., 4., -2., 1., 2.},
+                          I<index_type>{0, 2, 1, 3, 2, 1, 2, 3, 3},
+                          I<index_type>{0, 4, 6, 8, 9})},
+          u_sparse_inv{Csr::create(
+              exec, gko::dim<2>(4, 4),
+              I<value_type>{-.5, .125, .3125, .09375, .25, .125, -.5, .25, .5},
+              I<index_type>{0, 1, 2, 3, 1, 2, 2, 3, 3},
+              I<index_type>{0, 4, 6, 8, 9})}
```

Comment:
Yes, I do need it here as well because I create a CSR matrix, which uses the `create` function with perfect forwarding. Since template argument deduction fails / is illegal with braced init lists, I need to specify a type (and an `initializer_list` is ideal for that).
In the `ILU` case, we use [`matrix::initialize`](https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/matrix/dense.hpp#L604), which has `initializer_list` as a parameter type, so braced init lists work.